ddp-workspace/projects/ddp-sdk/src/lib/services/logging.service.ts

```diff
@@ -1,68 +1,20 @@
-import { Injectable, Inject, OnDestroy } from '@angular/core';
+import { Injectable, Inject } from '@angular/core';
 import { ConfigurationService } from './configuration.service';
-import { LogEvent } from '../models/logEvent';
 import { LogLevel } from '../models/logLevel';
-import { DdpException } from '../models/exceptions/ddpException';
-import { ExceptionDispatcher } from './exceptionHandling/exceptionDispatcher.service';
-import { Observable, Subscription, ReplaySubject } from 'rxjs';
-import { filter } from 'rxjs/operators';
 
 @Injectable()
-export class LoggingService implements OnDestroy {
-    private subjectSource = new ReplaySubject<LogEvent>();
-    private anchor: Subscription;
+export class LoggingService {
+    constructor(@Inject('ddp.config') private config: ConfigurationService) { }
 
-    constructor(
-        private dispatcher: ExceptionDispatcher,
-        @Inject('ddp.config') private config: ConfigurationService) {
-        this.initEventsListener();
-    }
-
-    public ngOnDestroy(): void {
-        this.anchor.unsubscribe();
-    }
-
-    public logDebug(source: string, event: any): void {
-        this.logMessage(LogLevel.Debug, source, event);
-    }
-
-    public logEvent(source: string, event: any): void {
-        this.logMessage(LogLevel.Info, source, event);
-    }
+    public logDebug = this.showEvent(LogLevel.Debug) ? console.debug.bind(window.console) : () => { };
 
-    public logWarning(source: string, event: any): void {
-        this.logMessage(LogLevel.Warning, source, event);
-    }
-
-    public logError(source: string, event: any): void {
-        this.logMessage(LogLevel.Error, source, event);
-    }
+    public logEvent = this.showEvent(LogLevel.Info) ? console.log.bind(window.console) : () => { };
 
-    public logException(source: string, exception: DdpException): void {
-        this.dispatcher.consume(exception);
-        this.logError(source, exception);
-    }
+    public logWarning = this.showEvent(LogLevel.Warning) ? console.warn.bind(window.console) : () => { };
 
-    public getSubscription(): Observable<LogEvent> {
-        return this.subjectSource.asObservable().pipe(
-            filter(event => event.level >= this.config.logLevel)
-        );
-    }
-
-    private initEventsListener(): void {
-        this.anchor = this.getSubscription().subscribe(event => {
-            switch (event.level) {
-                case LogLevel.Debug: console.debug(`${event.message}::${event.context}`); break;
-                case LogLevel.Info: console.log(`${event.message}::${event.context}`); break;
-                case LogLevel.Warning: console.warn(`${event.message}::${event.context}`); break;
-                case LogLevel.Error: console.error(`${event.message}::${event.context}`); break;
-            }
-        });
-    }
+    public logError = this.showEvent(LogLevel.Error) ? console.warn.bind(window.console) : () => { };
```

Comment:
I think you can use console.error instead of console.warn for this one.

Reply:
Good catch, thanks!